src/replica/duplication/duplication_sync_timer.h

```diff
@@ -73,6 +73,11 @@ class duplication_sync_timer
 
     std::vector<replica_ptr> get_all_replicas();
 
+    // duplication_sync_timer is async with replica close,so replica_duplicator_manager may already
+    // release
+    // dup should stop right now
+    bool replica_is_cloing_or_closed(gpid id);
```

Comment:
The same to https://github.com/apache/incubator-pegasus/pull/1608, you can judge the state of the replica by itself.